x-pack/libbeat/reader/parquet/parquet_test.go

```diff
@@ -218,17 +211,18 @@ func TestParquetWithFiles(t *testing.T) {
 
 // readJSONFromFile reads the json file and returns the data as an ordered map (row number -> json string)
 // along with the number of rows in the file
-func readJSONFromFile(t *testing.T, file *os.File) (map[int]string, int) {
+func readJSONFromFile(t *testing.T, filepath string) (map[int]string, int) {
+	fileBytes, err := os.ReadFile(filepath)
+	assert.NoError(t, err)
+	var rawMessages []json.RawMessage
+	err = json.Unmarshal(fileBytes, &rawMessages)
+	assert.NoError(t, err)
 	data := make(map[int]string)
-	scanner := bufio.NewScanner(file)
-	row := 0
-	for scanner.Scan() {
-		data[row] = scanner.Text()
+	var row int
+	for _, rawMsg := range rawMessages {
+		data[row] = string(rawMsg)
 		row++
 	}
```

Comment:
Rather than using `json.Unmarshal`, can we not use `json.Decoder` and stream in the objects? This would avoid the huge git diff that the current approach requires.

Reply:
@efd6 actually the test files were an ndjson before which gave a much smaller diff but Andrew suggested the json to be prettified hence this approach.

Reply:
No worries. Thanks.